### DISCUSSION TITLE
Listing Buckets Supabase Storage

### DISCUSSION THREAD

**Discussion by FallenDeity** [supabase/supabase#22139] [Questions] [answered]
So i create a few buckets in supabase storage from the dashboard and created policies with the following settings
![image](https://github.com/supabase/supabase/assets/61227305/4815684a-1b92-487d-97b0-582609c05e4b)
I also made the bucket public
Now when I try to list the buckets using `supabase-csharp` it works when I use my `service_role_key` but not when I use my `anon_key` is there a way to make it work with `anon`?

**Comment by GaryAustin1** [comment]
The policy you show should allow listing of the Avatars bucket with anon.  Basically you need select and you did not limit it to authenticated so should work.  This is to list the files in buckets.
If you are trying to list all of your buckets (not the files) then you need to add a select policy on the storage.buckets table (The UI also has an option for that.

**Comment by FallenDeity** [comment]
Ok so i added a policy under `storage.buckets`
![image](https://github.com/supabase/supabase/assets/61227305/2638aa9c-fe45-44d0-b7cd-c1c4a175efd8)
```cs
Client = new Supabase.Client(_supabaseUrl, _supabaseKey);
await Client.InitializeAsync();
var buckets = await Client.Storage.ListBuckets();
if (buckets is not null)
{
    System.Diagnostics.Debug.WriteLine("Buckets: " + buckets.Count);
    foreach (var bucket in buckets)
    {
        System.Diagnostics.Debug.WriteLine(bucket.Name);
    }
}
```
And here is my code when I use my anon key it gives me `Buckets: 0` its not fetching any buckets
And here is same code running with service key
```
Buckets: 3
Avatars
Playlists
Albums
```

Edit: I tried just listing files with anon key that seems to work but listing bucket with anon key doesn't.

**Reply by GaryAustin1** [reply, marked as answer]
The policy you added needs a USING clause with true.

**Reply by FallenDeity** [reply]
I see I'll try it tomorrow morning, Thank You ☺️.

**Reply by FallenDeity** [reply]
Thanks it works now!
